implementation/src/main/java/io/smallrye/mutiny/groups/MultiOnItem.java

```diff
@@ -520,4 +521,39 @@ public MultiOnItem(Multi<T> upstream) {
         return Infrastructure.onMultiCreation(new MultiScanOp<>(upstream, accumulator));
     }
 
+    /**
+     * Produces a new {@link Multi} invoking the given function when the current {@link Multi} fires an item (only once).
+     * The function transforms the received item into a failure that will be fired by the produced {@link Multi}.
+     * For asynchronous composition, see {@link #transformToUni(Function)}}.
+     *
+     * The upstream subscription is cancelled after the emission of the first item, as a failure is propagated downstream.
+     *
+     * @param mapper the mapper function, must not be {@code null}, must not return {@code null}
+     * @return the new {@link Multi}
+     */
+    public Multi<T> failWith(Function<? super T, ? extends Throwable> mapper) {
+        nonNull(mapper, "mapper");
+        return Infrastructure.onMultiCreation(transformToUniAndConcatenate(t -> {
+            Throwable failure = Objects.requireNonNull(mapper.apply(t), MAPPER_RETURNED_NULL);
```

Comment:
Why not our null-checks? (for consistency)

Reply:
I can use it, I will use the one throwing an NPE.

Reply:
If I use our null-check I cannot use the customize the error message and indicate that's the value produces by the function/supplier that is null.